src/cirrus/chef_tools.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+"""
+_chef_tools_
+
+Helpers/utils for modifying chef objects on the server and in a local repo
+
+"""
+import os
+import git
+import json
+from contextlib import contextmanager
+
+
+class ChefRepo(object):
+    """
+    _ChefRepo_
+
+    Wrapper for a git chef-repo to assist in
+    editing and committing changes to json files such as
+    environments and roles.
+
+    Provides contexts for editing envs etc and saving and
+    committing the changes.
+
+    Eg:
+
+    c = ChefRepo('./chef-repo')
+    with c.edit_environment('production') as env:
+        env['override_attributes']['thing'] = 'X.Y.Z'
+
+    """
+    def __init__(self, repo_dir, **options):
+        self.repo_dir = repo_dir
+        self.repo = git.Repo(repo_dir)
+        self.envs = options.get('environments_dir', 'environments')
+        self.roles = options.get('roles_dir', 'roles')
+
+    def checkout_and_pull(self, branch_name='master'):
+        """
+        _checkout_and_pull_
+
+        Check out the specified branch, git pull it
+
+        """
+        if self.repo.active_branch != branch_name:
+            dev_branch = getattr(self.repo.heads, branch_name)
+            dev_branch.checkout()
+
+        # pull branch_from from remote
+        ref = "refs/heads/{0}:refs/remotes/origin/{0}".format(branch_name)
+        return self.repo.remotes.origin.pull(ref)
+
+    def commit_files(self, commit_msg, *filenames):
+        """
+        _commit_files_
+
+        Add the list of filenames and commit them with the message provided
+        to the current branch in the repo specified.
+        Pushes changes to remote branch after commit
+
+        """
+        self.repo.index.add(filenames)
+        # commits with message
+        new_commit = self.repo.index.commit(commit_msg)
+        # push branch to origin
+        result = self.repo.remotes.origin.push(self.repo.head)
```

Comment:
Im a bit torn on what to do here TBH. I think I may refactor a bit so that it creates a local branch, commits and pushes that. Then the change can be PR'ed rather than robo-committed.